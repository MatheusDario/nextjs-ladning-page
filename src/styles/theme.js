export const theme = {
  colors: {
    white: '#f0f0f2',
    blue: '#023059',
    lightblue: '#065473',
    aquablue: '#08748c',
    green: '#02735e',
    lightgreen: '#6ab04c',
    lightgray: ' #cecece',
    red: '#dc143c',
    black: '#000000',
  },
  spacing: {
    xmall: '.8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
  },
  font: {
    sizes: {
      xmall: '.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
    },
    family: {
      default: "'Lato', sans-serif",
      title: "'Montserrat', sans-serif",
    },
  },
  breakpoints: {
    mobile: '@media (max-width:768px)',
    tests: '(max-width:768px)',
  },
};
