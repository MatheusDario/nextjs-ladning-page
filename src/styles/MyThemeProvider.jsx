import { ThemeProvider } from "styled-components";
import { theme } from "./theme";
import React from "react";
import { GlobalStyles } from "./global-styles";

export default function MyThemeProvider({ children }) {
  return (
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyles />
    </ThemeProvider>
  );
}

// MyThemeProvider.propTypes = {
//   children: P.node.isRequired,
// };
