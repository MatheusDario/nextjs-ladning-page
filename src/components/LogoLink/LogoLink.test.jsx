import { screen } from '@testing-library/react';
import { LogoLink } from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<LogoLink />', () => {
  it('should render logo text', () => {
    renderTheme(<LogoLink text="Logo" link="#home" />);
    screen.getByRole('heading', { name: 'Logo' });
    expect(screen.getByRole('link', { name: 'Logo' })).toHaveAttribute(
      'href',
      '#home',
    );
  });

  it('should render image logo', () => {
    renderTheme(<LogoLink text="Logo" link="#home" srcimg="logo.jpg" />);
    expect(screen.getByRole('img', { name: 'Logo' })).toHaveAttribute(
      'src',
      'logo.jpg',
    );
  });

  it('should match a snapshot', () => {
    const { container } = renderTheme(
      <LogoLink text="Logo" link="#home" srcimg="logo.jpg" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
