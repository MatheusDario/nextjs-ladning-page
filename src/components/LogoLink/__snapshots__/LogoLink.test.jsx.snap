// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should match a snapshot 1`] = `
.c1 {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
}

.c1 >img {
  max-height: 2.5rem;
}

.c0 {
  color: #023059;
  font-size: 1.6rem;
  text-transform: uppercase;
}

@media (max-width:768px) {
  .c1 {
    font-size: 1.6rem;
  }
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h1
  class="c0"
>
  <a
    class="c1"
    href="#home"
  >
    <img
      alt="Logo"
      src="logo.jpg"
    />
  </a>
</h1>
`;
